stetho-urlconnection/src/main/java/com/facebook/stetho/urlconnection/StethoURLConnectionManager.java

```diff
@@ -37,24 +31,37 @@
  */
 @NotThreadSafe
 public class StethoURLConnectionManager {
-  private static final AtomicInteger sSequenceNumberGenerator = new AtomicInteger(0);
+  private static boolean sIsStethoPresent = isStethoPresent();
+
+  private static boolean isStethoPresent() {
+    try {
+      Class.forName("com.facebook.stetho.Stetho");
+      return true;
+    } catch (ClassNotFoundException e) {
+      return false;
+    }
+  }
 
-  private final NetworkEventReporter mStethoHook = NetworkEventReporterImpl.get();
-  private final int mRequestId;
-  @Nullable private final String mFriendlyName;
+  private final Holder mHolder;
```

Comment:
You should add a comment that explains the purpose of the indirection through Holder.  Otherwise this looks like unnecessary complexity.